identity/src/main/java/com/stripe/android/identity/utils/NavigationUtils.kt

```diff
@@ -232,6 +229,10 @@ internal fun NavController.navigateUpAndSetArgForUploadFragment(): Boolean {
                 .build()
         )
     }
+    currentBackStackEntry?.destination?.id?.fragmentIdToRequirement()
```

Comment:
this is called when 
1. user clicks system back button
2. home button on top left
3. when an error screen shows up and user clicks "go back" on the error screen

`navigateUp` is called to pop an entry from backstack.

Current and previous entry would correspond to a type of data to collect, since we're recollecting that data, that type needs to be cleared.
E.g:
Before the function, backstack is
`[Consent screen, DocType screen, DocScan screen]`

After the function, backstack is
`[Consent screen, DocType screen]`

 `DocScan screen` is popped, `DocType screen` is shown - the user is about to select a new docType.

Therefore we need to clear data for both `DocScan screen`(currentBackStackEntry) and `DocType screen`(previousBackStackEntry)